cmake/find_cjson.cmake

```diff
@@ -1,21 +1,28 @@
+option(ATSDK_CJSON_PATH "Local cJSON path" OFF)
+option(ENABLE_CJSON_TEST "Test cJSON" OFF)
+set(FETCHCONTENT_TRY_FIND_PACKAGE_MODE OPT_IN) # only try find_package if FIND_PACKAGE_ARGS is set
+
 if(NOT TARGET cjson)
   # This installs the target cjson
   # Configurable variables
-  set(CJSON_BUILD_SHARED_LIBS OFF CACHE BOOL "Build cjson shared libraries")
-  option(ENABLE_CJSON_TEST "Enable tests for cjson" OFF)
-  set(FETCHCONTENT_TRY_FIND_PACKAGE_MODE OPT_IN) # only try find_package if FIND_PACKAGE_ARGS is set
 
   message(STATUS "[cJSON] fetching package...")
   include(FetchContent)
-
-  FetchContent_Declare(
-    cjson
-    URL https://github.com/DaveGamble/cJSON/archive/refs/tags/v1.7.17.zip
-    URL_HASH
-      SHA256=51f3b07aece8d1786e74b951fd92556506586cb36670741b6bfb79bf5d484216 # hash for v1.7.17 .zip release source code
-    # FIND_PACKAGE_ARGS 1.7.17 QUIET CONFIG
-  )
-
+  if(ATSDK_CJSON_PATH)
+    FetchContent_Declare(
+      cjson
+      SOURCE_DIR
+      ${CMAKE_SOURCE_DIR}/${ATSDK_CJSON_PATH}
```

Comment:
When ATSDK_CJSON_PATH is passed (and ATSDK_MBEDTLS_PATH for mbedtls) then cmake uses the relative path to the root CMakeLists.txt's directory.